infer/inference.cc

```diff
@@ -83,6 +83,22 @@ unique_ptr<cfg::CFG> Inference::run(core::Context ctx, unique_ptr<cfg::CFG> cfg)
     for (auto it = cfg->forwardsTopoSort.rbegin(); it != cfg->forwardsTopoSort.rend(); ++it) {
         cfg::BasicBlock *bb = *it;
         if (bb == cfg->deadBlock()) {
+            for (const auto &bind : bb->exprs) {
+                if (bind.value.isSynthetic() || bind.loc.empty()) {
+                    continue;
+                }
+
+                if (auto e = ctx.beginError(bind.loc, core::errors::Infer::DeadBranchInferencer)) {
+                    e.setHeader("This expression appears after an unconditional return");
+                    for (const auto &prevBasicBlock : bb->backEdges) {
+                        const auto &prevEnv = outEnvironments[prevBasicBlock->id];
+                        if (prevEnv.isDead) {
+                            continue;
+                        }
+                    }
```

Comment:
This loop doesn't appear to do anything?  Did you have some code here previously but decided to delete it?

Reply:
Ah yeah, at first I tried to print something like, "made dead because of this `return`" instruction or something like that but then I gave up on it.